lib/ecto/query/builder.ex

```diff
@@ -815,6 +831,34 @@ defmodule Ecto.Query.Builder do
   def field!(other),
     do: error!("expected atom in field/2, got: `#{inspect other}`")
 
+  defp escape_json_path(path) do
+    Enum.map(path, &quoted_json_path_element!/1)
+  end
+
+  defp quoted_json_path_element!({:^, _, [expr]}),
+    do: quote(do: Ecto.Query.Builder.json_path_element!(unquote(expr)))
```

Comment:
currently we simply interpolate the value into the query - if the value comes from user input it might be an SQL injection. (FWIW this is similar to how `field/2` works but given that one works on atoms, sql injections are less likely/impossible.)

 I've documented it as such, but I'm planning to see if I can handle this properly: interpolated values would be sent as params.

Reply:
AFAIK that's not true with `field` - we always quote the field names - https://github.com/elixir-ecto/ecto_sql/blob/wm-json-extract-path/lib/ecto/adapters/myxql/connection.ex#L463-L467

Reply:
Yup, good point. I've added proper escaping.